compositor_render/src/transformations/shader/node.rs

```diff
@@ -14,18 +14,27 @@ pub struct ShaderNode {
 }
 
 impl ShaderNode {
-    pub fn new(ctx: &WgpuCtx, shader: Arc<Shader>, params: &ShaderParam) -> Self {
+    pub fn new(ctx: &WgpuCtx, shader: Arc<Shader>, params: Option<&ShaderParam>) -> Self {
         // TODO: validation
-        // TODO: allow not passing custom parameters
-        let params = params.to_bytes();
 
-        let custom_params_buffer =
-            ctx.device
+        let custom_params_buffer = match params {
+            Some(params) => {
+                let params = params.to_bytes();
+                ctx.device
+                    .create_buffer_init(&wgpu::util::BufferInitDescriptor {
+                        label: Some("shader node custom params buffer"),
+                        usage: wgpu::BufferUsages::UNIFORM,
+                        contents: &params,
+                    })
+            }
+            None => ctx
+                .device
                 .create_buffer_init(&wgpu::util::BufferInitDescriptor {
-                    label: Some("shader node custom params buffer"),
+                    label: Some("shader node empty custom params buffer"),
+                    contents: &[0],
                     usage: wgpu::BufferUsages::UNIFORM,
-                    contents: &params,
-                });
+                }),
+        };
```

Comment:
With this, uniform in shader still has to be defined, right? I don't think defining `custom_params` (or sth like this) uniform in the shader when no parameters are used is a good solution. Maybe we should have two different bind groups layouts etc. for that.

Reply:
i was a bit surprised by this, but the shader works without defining this uniform

Reply:
also, [the spec](https://www.w3.org/TR/webgpu/#abstract-opdef-validating-gpuprogrammablestage) seems to confirm that shaders are allowed to not define bindings for some of the provided bind group entries.

This explains how bindings defined in a shader are validated:
> - For each binding that is [statically used](https://www.w3.org/TR/webgpu/#statically-used) by descriptor:
>    - [validating shader binding](https://www.w3.org/TR/webgpu/#abstract-opdef-validating-shader-binding)(binding, layout) must return true.

But the bindings that are not used by the shader are not validated, so they can be present and actually set to anything really

Reply:
Ok, I didn't know/noticed that :)